tdp/cli/commands/plan/edit.py

```diff
@@ -0,0 +1,214 @@
+        ]
+
+        temp_file = tempfile.NamedTemporaryFile(mode="w", suffix=".csv", delete=False)
+
+        header_text = f"""\
+# ------------------- DEPLOYMENT PLAN EDITOR --------------------------
+# Modify this file to edit the deployment plan.
+# Lines beginning with '#' are comments and will be ignored.
+#
+# Current plan id: {planned_deployment_log.id}
+# Temporary file: {temp_file.name}
+#
+# Execution Order:
+# - Operations are executed in the order they appear below.
+# - You can add, remove, or rearrange operations.
+# - Use `tdp nodes` command to view a list of available operations.
+#
+# Operations options:
+# - Specify a host with the `on` keyword.
+#   Example: hdfs_datanode_config on worker_02
+# - Add extra variables using the `with` keyword and separate with `,`.
+#   Example: wait_sleep with rolling-interval=5, mock-arg=example
+# ------------------- DEPLOYMENT PLAN EDITOR --------------------------
+"""
+
+        temp_file.write(header_text + "\n".join(operation_lines))
+        temp_file.close()
+
+        click.echo(
+            f"Current plan id: {planned_deployment_log.id}\nTemporary file : {temp_file.name}"
+        )
+
+        try:
+            program_exit = False
+            # Loop until plan has no error or if user wants to stop editing the plan
+            while not program_exit:
+                # Loop until user wants to save editing the plan
+                program_continue = False
+                while not program_continue:
+                    run(
+                        [
+                            os.environ.get("VISUAL", "")
+                            or os.environ.get("EDITOR", "")
+                            or "vi",
+                            temp_file.name,
+                        ]
+                    )
+
+                    # Workaround to wait for the editor like vscode to be closed
+                    click.echo(
+                        message="Press 'e' to reopen the editor, or another key to save the plan : ",
+                        nl=False,
+                    )
+                    user_choice = click.getchar(echo=True)
+                    # Print a newline
+                    click.echo()
+                    if user_choice.lower() != "e":
+                        program_continue = True
+
+                try:
+                    with open(temp_file.name, "r") as temp_file:
+                        file_content = temp_file.read()
+                except FileNotFoundError as e:
+                    raise click.ClickException(str(e)) from e
+
+                try:
+                    # Remove empty elements and comments
+                    # and get the operations, hosts and extra vars in a list
+                    new_operations_hosts_vars = [
+                        _parse_line(line)
+                        for line in file_content.split("\n")
+                        if line and not line.startswith("#")
+                    ]
```

Comment:
Maybe we can move this to a `_parse_file` function. This could be usefull later if we want to create a plan from a file.

Reply:
Done, I created the method